### PR TITLE
Added Statistics Page

### PR DESCRIPTION
**Pull request by GamingMaster2000**
## Description

Added a statistics page (location at /stats) that can show the number of Pokemon seen over a certain time period. Pokemon can be sorted by how often they've been seen, Pokemon id number or name. The page auto-refreshes to get up to date info.

EDIT: I closed my old PR and opened this one because being new to Github I did something wrong :p

## Motivation and Context

Gives us a way to see how often things appear on our maps

## How Has This Been Tested?

Tested on my own server

## Screenshots (if appropriate):
stats_page

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by willgrnr** [comment]
If you'd like, tonight I can pretty it up with bootstrap to match the current map page. I'll make a pull request to your fork.

**Comment by GamingMaster2000** [comment]
Sure, if you want that'd be great. Thanks

**Comment by BenVlodgi** [comment]
Could you by any chance click on those pokemon and get a list of location/time they appeared?

**Comment by GamingMaster2000** [comment]
Not yet but that's something that I have in mind. :)

**Comment by GamingMaster2000** [comment]
Just added some more information for the Pokemon seen. The main page will show the time and location of the last of that Pokemon to be seen. Clicking 'More locations' should show all the locations & times that Pokemon has been seen from.

**Comment by marauder37** [comment]
Have you put a link to this stats page from anywhere else? Note that we just merged a slide-out stats panel, maybe you could add a "full stats" link on that panel

**Comment by GamingMaster2000** [comment]
I'll add a full stats link now.

**Comment by GamingMaster2000** [comment]
Ok, I've added a "Full Stats" link under the Pokemon list on the stats slide-out since at the moment the full stats only shows Pokemon stats. We can move/change it should the /stats page add more later on.

**Comment by marauder37** [comment]
It's a bit rough around the edges but I'm +1 on this

**Comment by marauder37** [comment]
Other reviewers feel this needs more features before merging, since we have the stats panel now. Is there anything else you were thinking of adding?

**Comment by reyqn** [comment]
@marauder37 I feel like those statistics are much more useful than the ones currently implemented.
@GamingMaster2000 I've got this issue. Any clue?
screenfetch-2016-07-28_22-19-41
Pokemons are disappearing for all durations (but present in lifetime where NaN is everywhere)

**Comment by GamingMaster2000** [comment]
@marauder37 I've thinking about adding in location based filtering as well so we can see the odds of Pokemon appearing in a certain area rather than world wide and possibly changing the way it displays information but probably nothing that would count as significant new features. If that's still not enough just let me know (and maybe close this PR?) so I don't waste more time on this page if it's not enough to be merged.

@reyqn I'll look into it and see what I can find out (assuming this PR continues).

**Comment by reyqn** [comment]
Yup this fixes Na times. However most frequently seen Pokemon are still missing for me. I'll go through the code tomorrow.

**Comment by GamingMaster2000** [comment]
I think I've got a fix for the missing Pokemon, i'm just doing some more testing before pushing a commit :)

**Comment by reyqn** [comment]
Oh okay. Then I can sleep in peace. Thanks for your work :+1:

**Comment by marauder37** [comment]
If you get it working I'll merge it

**Comment by GamingMaster2000** [comment]
@marauder37 I think it's working correctly now but could do with a fresh set of eyes to check for me.

**Comment by reyqn** [comment]
Yup. At least working for me now :+1:

**Comment by AHAAAAAAA** [comment]
Tested, works as intended but could use quite a bit of UI sprucing. For example, clicking on 'More locations' creates a window with a basic list but if you do this at the bottom of the page, you will not see it as it only appears at the very top of the page. I don't know if we should just merge and have a separate PR to spruce it up or wait here for somebody to step in and spruce it up. I am unable to tackle this right now personally.

**Comment by GamingMaster2000** [comment]
@AHAAAAAAA I'm working on replacing that appearances list with an instance of Google Maps that will show where & when the Pokemon have appeared. I'm not really sure if I'm capable of handling a heat map type situation but it should at least be able to form the basis for it. Should help things look a little better.

File: templates/map.html [hunk 1]
```diff
@@ -164,6 +164,7 @@
 			<center><h1 id="stats-pkmn-label"></h1></center>
 		  </div>
 		  <div id="pokemonList" style="color: black;"></div>
+          <div><a href="/stats">Full Stats</a></div>
```
Comment (tkSimon) [inline templates/map.html]: Please don't use links to the document root. Use `stats` instead of `/stats`

**Comment by willgrnr** [comment]
I'm still going to give this a makeover, just haven't had the time. Up north without a computer currently. I'll have a 2 day window this week to do some work otherwise it will have to wait until the week after. 

If you want to merge this and open a PR for a design you can assign me to it.

**Comment by GamingMaster2000** [comment]
Today I added a heat map to the additional info/all locations/overlay section. I've also improved the UI a bit. Basically it should now all look like this:

Main page:
main

Options:
options

Overlay:
overlay

Overlay details (when hovering/clicking on a marker):
times

**Comment by AHAAAAAAA** [comment]
Beautiful.

**Pull request merged**